lua/resession/init.lua

```diff
@@ -69,11 +81,47 @@ M.get_current = function()
   return tab_sessions[tabpage] or current_session
 end
 
----Detach from the current session
-M.detach = function()
-  current_session = nil
-  local tabpage = vim.api.nvim_get_current_tabpage()
-  tab_sessions[tabpage] = nil
+---Detach from a session
+---@param name? string
+---@param opts? resession.DetachOpts
+---@param target_tabpage? integer
+M.detach = function(name, opts, target_tabpage)
+  opts = vim.tbl_extend("keep", opts or {}, {
+    notify = true,
+  })
+  if target_tabpage then
+    name = tab_sessions[target_tabpage]
+    if not name then
+      error(string.format("No session attached to tab %s", target_tabpage))
+    end
+  elseif name then
+    for tabnr, session in pairs(tab_sessions) do
+      if name == session then
+        target_tabpage = tabnr
+        break
+      end
+    end
+    if name ~= current_session and not target_tabpage then
+      error(string.format('No session with name "%s"', name))
+    end
+  else
+    name = M.get_current()
+    target_tabpage = vim.api.nvim_get_current_tabpage()
+  end
+  if name or tab_sessions[target_tabpage] then
+    current_session = nil
+    if target_tabpage then
+      tab_sessions[target_tabpage] = nil
+    end
+    if opts.notify then
+      vim.notify(string.format('Detached from session "%s"', name))
+    end
+    dispatch("post_detach", name, opts, target_tabpage)
```

Comment:
We don't need to pass in the entire `opts` here. The only information that is needed is the session name and the tabpage, but the tabpage should _only_ be included if the deleted session was a tab-scoped session. The reason we pass `opts` in the other hooks is because they contain more than just UI logic (mainly, the `dir`).